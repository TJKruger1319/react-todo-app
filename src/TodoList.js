import React, { useState } from "react";
import { Todo, NewTodoForm} from "./NewTodoForm";
import { v4 as uuid } from "uuid";

function TodoList() {
    const [todos, setTodo] = useState([]);

    const removeTodo = id => {
        setTodo(todos.filter(todo => todo.id !== id));
    };

    const addTodo = todo => {
        let newTodo = {...todo, id: uuid() };
        setTodo(todos => [...todos, newTodo]);
    };

    return (
        <div>
            <NewTodoForm addTodo={addTodo}/>
            <ul>
                {todos.map(({ id, text }) =>
                <li key={id}>
                    <Todo 
                    key={id} 
                    id={id} 
                    text={text}   
                    remove={removeTodo}
                    />
                </li>
                )}
            </ul>
        </div>
    )
}

export default TodoList;