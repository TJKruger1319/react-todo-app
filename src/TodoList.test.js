import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("Can add a new todo", function() {
    const { getByLabelText, queryByText } = render(<TodoList />);

    expect(queryByText("todo1")).not.toBeInTheDocument();

    const textInput = getByLabelText("Text");
    const submitButton = queryByText("Add new todo");

    fireEvent.change(textInput, { target: { value: "todo1"}});
    fireEvent.click(submitButton);

    expect(queryByText("todo1")).toBeInTheDocument();
})

it("matches snapshot", function() {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
  });