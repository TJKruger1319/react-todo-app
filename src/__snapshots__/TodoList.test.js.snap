// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="text"
      >
        Text
      </label>
      <input
        id="text"
        name="text"
        placeholder="text"
        type="text"
        value=""
      />
      <button>
        Add new todo
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
